markdown.dtx

```diff
@@ -28024,23 +28044,30 @@ function M.reader.new(writer, options)
       else
         char_length = pos + 1
       end
-      c = lpeg.match({ C(utf8_by_byte_count(char_length)) },s,i+pos)
-      if (c ~= nil) and (unicode.utf8.match(c, chartype)) then
-        return i
+
+      if (chartype == "punctuation") then
+        if lpeg.match(punctuation_by_size[char_length], s, i+pos) then
```

Comment:
Please, say more about this. The pattern `punctuation_3_byte_ungrouped` seems a burden from readability and maintenance perspective and it's unclear to me why we need it. What is the difference between `unicode.utf8.match(lpeg.match({ C(utf8_by_byte_count(char_length)) },s,i+pos), chartype)` and `lpeg.match(punctuation_by_size[char_length], s, i+pos)`, functionally?

Reply:
As discussed with @lostenderman on Discord on 2023-03-21, we need a special category, because CommonMark's punctuation includes the Unicode categories `P` (Punctuation), `S` (Symbol), as well as other ASCII characters:

- https://spec.commonmark.org/0.31.2/#ascii-punctuation-character
- https://spec.commonmark.org/0.31.2/#unicode-punctuation-character

Since Selene Unicode does not support matching the category `S`, using LPEG patterns seems as the cleanest solution. Using a separate name (`punctuation`) instead of `%p` is also justified. However, all this should be documented in the code.